app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingViewModel.kt

```diff
@@ -756,11 +788,27 @@ class VaultItemListingViewModel @Inject constructor(
 
     private fun handleDismissFido2ErrorDialogClick() {
         clearDialogState()
-        sendEvent(
-            VaultItemListingEvent.CompleteFido2Registration(
-                result = Fido2RegisterCredentialResult.Error,
-            ),
-        )
+        when {
+            state.fido2CredentialRequest != null ||
+                state.fido2CredentialAssertionRequest != null -> {
```

Comment:
I'm confused about this - the registration and assertion requisition both trigger the same event? If so, could we rename that event so it doesn't have `Registration` in the name?

Reply:
They do not. That logic was erroneous. There are different completion events for each operation. Good catch!